controllers/provisioners/eks/scaling/launchtemplate.go

```diff
@@ -243,6 +246,41 @@ func (lt *LaunchTemplate) Drifted(input *CreateConfigurationInput) bool {
 		drift = true
 	}
 
+	if len(latestVersion.LaunchTemplateData.LicenseSpecifications) != len(input.LicenseSpecifications) {
```

Comment:
It might be eaiser to create a `ec2.LaunchTemplateLicenseConfiguration` object from `input.LicenseSpecifications`, make sure both are sorted, and then just do a reflect.DeepEqual() - same as how we do Volumes - this will also allow us to remove the len check.